packages/flutter_tools/lib/src/vscode/vscode.dart

```diff
@@ -140,26 +147,26 @@ class VsCode {
   //   $HOME/.vscode/extensions
   //   $HOME/.vscode-insiders/extensions
   static List<VsCode> _installedLinux() {
-    return _findInstalled(
-      <String, String>{'/usr/share/code': '.vscode'},
-      <String, String>{'/usr/share/code-insiders': '.vscode-insiders'}
-    );
+    return _findInstalled(<_VsCodeInstallLocation>[
+      new _VsCodeInstallLocation('/usr/share/code', '.vscode'),
+      new _VsCodeInstallLocation('/usr/share/code-insiders', '.vscode-insiders', isInsiders: true)
+    ]);
   }
 
   static List<VsCode> _findInstalled(
-      Map<String, String> stable, Map<String, String> insiders) {
-    final Map<String, String> allPaths = <String, String>{};
-    allPaths.addAll(stable);
-    if (_includeInsiders)
-      allPaths.addAll(insiders);
+      List<_VsCodeInstallLocation> allLocations) {
+    final List<_VsCodeInstallLocation> searchLocations = 
```

Comment:
This could be `Interable<_VsCodeInstallLocation>`, which would save you the `toList()`

Reply:
And would've saved me the failed build on the first one because I'd run the tests without `-c`! ;(

Changed now.